### PR TITLE
Feature/32835 classes for ad addresses 2

### PR DESCRIPTION
**Pull request by jgj-magenta-aps**
Jeg er bare ikke gode venner med git :-)

### CODE REVIEW

File: mox_stsorgsync/os2mo.py [hunk 1]
```diff
@@ -70,15 +70,58 @@ def user_uuids():
...
+            if (
+                candidate["address_type"]["uuid"] == cls
+                and candidate.get("visibility",
+                                  {"scope": "PUBLIC"})["scope"] == "PUBLIC"
+            ):
+                chosen = {"Value": candidate["name"],
+                          "Uuid": candidate["uuid"]}
+
+    if not prioritized_classes and len(candidates):
+        for candidate in reversed(candidates):
```
Comment (cmoesgaard) [inline mox_stsorgsync/os2mo.py:89]: Why do we reverse the list of candidates here?

File: tests/test_os2mo.py [hunk 2]
```diff
@@ -6,30 +6,130 @@
...
         user = {}
         os2mo.addresses_to_user(user, [
             {'address_type': {'scope': "DAR"}, 'name': 'ABC', 'uuid': "XYZ"}
         ])
         self.assertEqual(user, {'Location': {'Uuid': 'XYZ', 'Value': 'ABC'}})
 
+    def test_secondary_address_when_first_is_secret(self):
+        "only the classes given are chosen and only of PUBLIC or no visibility"
+        user = {}
+        os2mo.settings["STSORGSYNC_PHONE_SCOPE_CLASSES"] = [
```
Comment (cmoesgaard) [inline tests/test_os2mo.py:37]: Consider using `patch` for temporarily overriding settings.

File: mox_stsorgsync/os2mo.py (hunk 1 above)
Comment (jgj-magenta-aps) [reply mox_stsorgsync/os2mo.py:89]: Previous behaviour was to overwrite until no more addresses came. In order to keep that behaviour I reversed the list.

File: tests/test_os2mo.py (hunk 2 above)
Comment (jgj-magenta-aps) [reply tests/test_os2mo.py:37]: I shall!

**Pull request closed without merging**